webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +95,1093 @@ public WebSiteSessionController createComponentSessionController(
+      String id = scc.getNextId();
+
+      // Persist uploaded website inside database, type=2
+      SiteDetail descriptionSite =
+          new SiteDetail(id, scc.getComponentId(), nomSite, description, nomPage, 2, null, null,
+              0, popup);
+
+      descriptionSite.setPositions(positions);
+      int result = scc.createWebSiteFromZipFile(descriptionSite, fileItem);
+      switch (result) {
+        case 0:
+          createSite(scc, descriptionSite, listeIcones, listeTopics);
+          Collection<SiteDetail> listeSites = scc.getAllWebSite();
+          request.setAttribute("ListSites", listeSites);
+          request.setAttribute("BookmarkMode", Boolean.valueOf(scc.isBookmarkMode()));
+          destination = "/webSites/jsp/manage.jsp";
+          break;
+        case -1:
+          // le nom de la page principale n'est pas bonne, on supprime ce qu'on a dezipe
+          scc.deleteDirectory(scc.getWebSitePathById(descriptionSite.getId()));
+          request.setAttribute("Site", descriptionSite);
+          request.setAttribute("AllIcons", scc.getAllIcons());
+          request.setAttribute(ICON_LIST_PARAM, listeIcones);
+          request.setAttribute("UploadOk", Boolean.TRUE);
+          request.setAttribute("SearchOk", Boolean.FALSE);
+          destination = "/webSites/jsp/descUpload.jsp";
+          break;
+        case -2:
+          request.setAttribute("Site", descriptionSite);
+          request.setAttribute("AllIcons", scc.getAllIcons());
+          request.setAttribute(ICON_LIST_PARAM, listeIcones);
+          request.setAttribute("UploadOk", Boolean.FALSE);
+          destination = "/webSites/jsp/descUpload.jsp";
+          break;
+        case -3: // the zip content isn't correct
+          scc.deleteDirectory(scc.getWebSitePathById(descriptionSite.getId()));
+          request.setAttribute("Site", descriptionSite);
+          request.setAttribute("AllIcons", scc.getAllIcons());
+          request.setAttribute(ICON_LIST_PARAM, listeIcones);
+          request.setAttribute("UploadOk", Boolean.FALSE);
+          destination = "/webSites/jsp/descUpload.jsp";
+          break;
+      }
+    }
+    return destination;
+  }
 
-        if (action == null) {
-          id = "0";
-          action = "Search";
+  private void createSite(final WebSiteSessionController scc, SiteDetail site, String listOfIcons,
+      String listOfTopics) throws WebSitesException {
+        /* creation en BD */
+    ArrayList<String> listIcons = new ArrayList<>();
+    if (listOfIcons != null) {
+      for (String icon : listOfIcons.split(",")) {
+        if (!icon.isEmpty()) {
+          listIcons.add(icon);
         }
+      }
+    }
 
-        FolderDetail webSitesCurrentFolder = scc.getFolder(id);
-        scc.setSessionTopic(webSitesCurrentFolder);
-        request.setAttribute("CurrentFolder", webSitesCurrentFolder);
+    String pubId = scc.createWebSite(site);
+    if (listIcons.size() > 0) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1155)